include/boost/url/bnf/parse.hpp

```diff
@@ -16,58 +16,77 @@
 #include <boost/type_traits/make_void.hpp>
 #include <type_traits>
 
+/** @file
+
+   Parsing commands
+
+   This file contains the basic overloads for parsing BNF elements
+
+   @par Design of BNF elements
+
+   A BNF element implements the rules for parsing a BNF grammar rule and stores parsing results.
+
+   A BNF element:
+
+    - Is default constructible
+    - Holds the result, usually as a reference, of parsing
+    - Has a nested value_type
+    - Has `value_type& operator*() const noexcept`
+
+   Users should define a free function `parse` as a customization point defining how to parse their
+   BNF elements as part of the same architecture that might include arbitrary BNF elements in expressions.
+
+   These new function overloads may be defined in other namespaces. As with `std::swap`, the design relies
+   on Argument Dependent Lookup (ADL) to find these overloads.
+
+*/
+
+
 namespace boost {
 namespace urls {
 namespace bnf {
 
-/*  Design
-
-    - An element is default constructible
-    - Holds the result of parsing
-    - overloads the free function `parse`
-    - has a nested value_type
-    - has `value_type& operator*() const noexcept`
-*/
 
-/** @brief Parse a literal character
+/** Parse a substring as a BNF literal character
```

Comment:
you don't need to say "BNF" here, just "Parse a literal character"

Reply:
OK. I removed it from all functions.